docs/src/extend/custom-rule-plugin-tutorial.md

```diff
@@ -0,0 +1,431 @@
+---
+title: Custom Rule & Plugin Tutorial
+eleventyNavigation:
+    key: custom rule plugin tutorial
+    parent: create plugins
+    title: Custom Rule & Plugin Tutorial
+    order: 1
+---
+This tutorial covers how to create a custom rule and plugin for ESLint.
+
+You can create custom rules to validate if your code meets a certain expectation, and determine what to do if it does not meet that expectation. Plugins package custom rules and other configuration, allowing you to easily share and reuse them in different projects.
+
+To learn more about custom rules and plugins refer to the following documentation:
+
+* [Custom Rules](custom-rules)
+* [Plugins](plugins)
+
+## Why Create a Custom Rule?
+
+Create a custom rule if the ESLint [built-in rules](../rules/) and community-published custom rules do not meet your needs.
```

Comment:
Let's expand on this a bit.

```suggestion
Create a custom rule if the ESLint [built-in rules](../rules/) and community-published custom rules do not meet your needs. You might create a custom rule to enforce a best practice for your company or project, to prevent a particular bug from recurring, or to ensure compliance with a style guide.   
```